qutebrowser/completion/models/histcategory.py

```diff
@@ -47,27 +47,6 @@ def __init__(self, *,
         self.delete_func = delete_func
         self._empty_prefix = None  # type: typing.Optional[str]
 
-    def _atime_expr(self):
-        """If max_items is set, return an expression to limit the query."""
-        max_items = config.val.completion.web_history.max_items
-        # HistoryCategory should not be added to the completion in that case.
-        assert max_items != 0
-
-        if max_items < 0:
-            return ''
-
-        min_atime = sql.Query(' '.join([
-            'SELECT min(last_atime) FROM',
-            '(SELECT last_atime FROM CompletionHistory',
-            'ORDER BY last_atime DESC LIMIT :limit)',
-        ])).run(limit=max_items).value()
-
-        if not min_atime:
-            # if there are no history items, min_atime may be '' (issue #2849)
-            return ''
-
-        return "AND last_atime >= {}".format(min_atime)
```

Comment:
Now I cant remember why we had this as opposed to just a `LIMIT`. The latter seems way simpler :confused:

Reply:
Introduced in https://github.com/rcorre/qutebrowser/commit/c32d452786508c9bc7f27f1e8c660001b310eaa9. Still can't remember why I did it that way. A simple `LIMIT` seems better.

Reply:
I looked into this (including the commit you reference) and while the way I did it is simpler, it's not really equivalent, although I believe it's the right approach.

The old way would limit the total number of entries to `completion.web_history.max_items`, and then it would filter to entries matching your pattern, resulting in always a lower number than the limit.

Now we filter by the pattern and then limit, so you always end up with `completion.web_history.max_items` items assuming you have enough entries.